composer/algorithms/alibi/attention_surgery_functions/_bert.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+from typing import Optional, Tuple
+from types import MethodType
+
+import torch
+from torch import nn
+
+from composer.algorithms.alibi.attention_surgery_functions.utils import (
+    register_surgery_function_builder,
+    register_alibi,
+    zero_and_freeze_expand_position_embeddings
+)
+from transformers.models.bert.modeling_bert import BertSelfAttention, BertEmbeddings
+from transformers.models.roberta.modeling_roberta import RobertaSelfAttention, RobertaEmbeddings
+
+@register_surgery_function_builder(BertEmbeddings, RobertaEmbeddings)
+def build_bert_embedding_converter(max_sequence_length: int):
+    """Builds a function to remove positional embeddings.
+    
+    The built function also expands `position_ids` buffer to support `max_sequence_length` tokens.
+    """
+    def convert_position_embeddings_and_ids(module: torch.nn.Module, module_index: Optional[int] = None):
+        del module_index # unused
+        zero_and_freeze_expand_position_embeddings(
+            module, max_sequence_length, position_embedding_attribute='position_embeddings')
+
+        module_device = next(module.parameters()).device
+        module.register_buffer(
+            "position_ids", torch.arange(max_sequence_length).expand((1, -1)).to(module_device))
+        return module
+    return convert_position_embeddings_and_ids
+
+
+@register_surgery_function_builder(BertSelfAttention, RobertaSelfAttention)
+def build_bert_attention_converter(max_sequence_length: int):
+    """Builds a function that does model surgery to add ALiBi to Bert-style SelfAttention."""
+    def convert_attention(module: torch.nn.Module, module_index: Optional[int] = None):
```

Comment:
Same as above

Reply:
I actually had to change the type signatures here to address issues with pyright. For example, `convert_attention` will still use `module: torch.nn.Module`, but I added asserts throughout to make sure that `module` is always an instance of one of the classes in the decorator.